### PR TITLE
GR: implement De-escalation

### PR DESCRIPTION
**Pull request by strib**

### CODE REVIEW

File: server/game/cards/07-GR/DeEscalation.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        this.play({
+            effect: 'destroy all creatures{1}',
+            effectArgs: (context) => [
+                context.player.opponent
+                    ? ' and have ' +
+                      context.player.opponent.name +
+                      ' archive the top 3 cards of their deck'
+                    : ''
+            ],
+            gameAction: [
```
Comment (romanogit) [inline server/game/cards/07-GR/DeEscalation.js:16]: this one needs to be sequential. If opponent has a Rad Penny, for example, they must be able to archive it.

**Review by romanogit — CHANGES_REQUESTED** [review]

File: server/game/cards/07-GR/DeEscalation.js (hunk 1 above)
Comment (strib) [reply server/game/cards/07-GR/DeEscalation.js]: Good catch!  Fixed, thanks.

File: server/game/cards/07-GR/DeEscalation.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            effectArgs: (context) => [
+                context.player.opponent
+                    ? ' and have ' +
+                      context.player.opponent.name +
+                      ' archive the top 3 cards of their deck'
+                    : ''
+            ],
+            gameAction: ability.actions.sequential([
+                ability.actions.destroy((context) => ({ target: context.game.creaturesInPlay })),
+                ability.actions.conditional((context) => ({
```
Comment (romanogit) [inline server/game/cards/07-GR/DeEscalation.js:18]: I do not think you need to add conditional here.

```javascript
gameAction: ability.actions.sequential([
                ability.actions.destroy((context) => ({ target: context.game.creaturesInPlay })),
                ability.actions.archive({
                   target: context.player.opponent ? context.player.opponent.deck.slice(0, 3) : []
                })
])
```

File: server/game/cards/07-GR/DeEscalation.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                    condition: context.player.opponent && context.player.opponent.deck.length > 0,
+                    trueGameAction: ability.actions.archive({
+                        target: context.player.opponent.deck.slice(
+                            0,
+                            Math.min(context.player.deck.length, 3)
```
Comment (romanogit) [inline server/game/cards/07-GR/DeEscalation.js:23]: Should be context.player.opponent.deck.length

No need to worry as slice() returns less than 3 if array is small.
```
> [1,2].slice(0,3)
[ 1, 2 ]
> 
```

**Review by romanogit — CHANGES_REQUESTED** [review]

**Review by romanogit — CHANGES_REQUESTED** [review]
I suggested some simplifications.

File: server/game/cards/07-GR/DeEscalation.js (hunk 3 above)
Comment (strib) [reply server/game/cards/07-GR/DeEscalation.js]: Done, thanks.

File: server/game/cards/07-GR/DeEscalation.js (hunk 2 above)
Comment (strib) [reply server/game/cards/07-GR/DeEscalation.js]: Done, thanks.

**Review by romanogit — APPROVED** [review]

**Pull request merged**